models/meshmodel/core/v1alpha2/relationship.go

```diff
@@ -0,0 +1,69 @@
+package v1alpha2
+
+import (
+	"fmt"
+	"path/filepath"
+
+	"github.com/google/uuid"
+	"github.com/layer5io/meshkit/database"
+	"github.com/layer5io/meshkit/models/meshmodel/core/v1beta1"
+	"github.com/layer5io/meshkit/models/meshmodel/entity"
+	"github.com/layer5io/meshkit/utils"
+	"gorm.io/gorm/clause"
+)
+
+const SchemaVersion = "core.meshery.io/v1alpha2"
+
+type RelationshipDefinition struct {
+	ID uuid.UUID `json:"id"`
+	v1beta1.VersionMeta
+	Kind string `json:"kind,omitempty" yaml:"kind"`
+	// The property has been named RelationshipType instead of Type to avoid collision from Type() function, which enables support for dynamic type.
+	// Though, the column name and the json representation is "type".
+	RelationshipType string                 `json:"type" yaml:"type" gorm:"type"`
+	SubType          string                 `json:"subType" yaml:"subType"`
+	EvaluationQuery  string                 `json:"evaluationQuery" yaml:"evaluationQuery" gorm:"evaluationQuery"`
+	Metadata         map[string]interface{} `json:"metadata"  yaml:"metadata" gorm:"type:bytes;serializer:json"`
+	ModelID          uuid.UUID              `json:"-" gorm:"index:idx_relationship_definition_dbs_model_id,column:model_id"`
+	Model            v1beta1.Model          `json:"model" gorm:"foreignKey:ModelID;references:ID"`
+	Selectors       []map[string]interface{} `json:"selectors"  yaml:"selectors" gorm:"type:bytes;serializer:json"`
+}
+
+func (r RelationshipDefinition) TableName() string {
+	return "relationship_definition_dbs"
+}
+
+func (r RelationshipDefinition) Type() entity.EntityType {
+	return entity.RelationshipDefinition
+}
+func (r RelationshipDefinition) GetID() uuid.UUID {
+	return r.ID
+}
+
+func (r *RelationshipDefinition) GetEntityDetail() string {
+	return fmt.Sprintf("type: %s, definition version: %s, kind: %s, model: %s, version: %s", r.Type(), r.Version, r.Kind, r.Model.Name, r.Model.Version)
+}
+
+func (r *RelationshipDefinition) Create(db *database.Handler, hostID uuid.UUID) (uuid.UUID, error) {
```

Comment:
Why is there no Swagger / API docs?

Reply:
This is a DB functions, the swagger docs are present for the handles inside `meshery/meshery`